build.sbt

```diff
@@ -268,28 +272,30 @@ def buildScalacOptions(scalaVersion: String) = {
     case _ => Seq.empty
   }
 
-  val javaVersionSpecificOptions =
-    if (scala.util.Properties.isJavaAtLeast("9"))
-      Seq("-release", "8") // ensure Java backwards compatibility (DAFFODIL-2579)
-    else
-      Seq.empty
+  commonOptions ++ scalaVersionSpecificOptions
```

Comment:
Added per comment. 

But... It's a bit trickier. Scala 2.12 can't specify a release other than 8. So we can't just default to specifying java 11 class files when we are using Java 21, because scala warns about that, and we're fatal warnings. 

So I did the minSupportedJavaVersion thing, but it's only used by java compilation for now. We have very small amounts of java code today, but those would complain about java 8, so those will use Java 11 classes IF the JVM is Java 21 only. Scala compilation does not specify any "-target" option.